CertificateRetrievingAndCreation.php

```diff
@@ -0,0 +1,481 @@
+
+    function __construct()
+    {
+        $this->client = new Client();
+    }
+
+    public function fetchApi(): void
+    {
+        $issuePage = 1;
+        $issuesToFetchPerPage = 20;
+        $fetchedMatchingIssue = false;
+
+        do {
+            $allissues = $this->client->api('issue')->all('codecheckers', 'register', [
+                'state'     => 'all',          // 'open', 'closed', or 'all'
+                'labels'    => 'id assigned',  // label
+                'sort'      => 'updated',
+                'direction' => 'desc',
+                'per_page'  => $issuesToFetchPerPage, // issues that will be fetched per page
+                'page'      => $issuePage,
+            ]);
+
+            // stop looping if no more issues exist and we haven't yet found a matching issue
+            if (empty($allissues) && empty($this->issue)) {
+                throw new NoMatchingIssuesFoundException("There was no Issue found with a '|' inside the GitHub Codecheck Register.");
+            }
+
+            foreach ($allissues as $issue) {
+                if (strpos($issue['title'], '|') !== false) {
+                    $this->issues[] = $issue;
+                    $fetchedMatchingIssue = true;
+                }
+            }
+
+            $issuePage++;
+        } while (!$fetchedMatchingIssue);
+    }
+
+    public function addIssue(
+        CertificateIdentifier $certificateIdentifier,
+        string $codecheckVenueType,
+        string $codecheckVenueName
+    ): void {
+        $token = $_ENV['CODECHECK_REGISTER_GITHUB_TOKEN'];
+
+        $this->client->authenticate($token, null, Client::AUTH_ACCESS_TOKEN);
+
+        $repositoryOwner = 'codecheckers';
+        $repositoryName = 'testing-dev-register';
+        $issueTitle = 'New CODECHECK | ' . $certificateIdentifier->toStr();
+        $issueBody = '';
+        $labels = ['id assigned'];
+
+        $labels[] = $codecheckVenueType;
+        $labels[] = $codecheckVenueName;
+
+        $this->client->api('issue')->create(
+            $repositoryOwner,
+            $repositoryName,
+            [
+                'title' => $issueTitle,
+                'body'  => $issueBody,
+                'labels' => $labels
+            ]
+        );
+    }
+
+    public function getIssues(): array
+    {
+        return $this->issues;
+    }
+}
+
+
+// CODECHECK GitHub Issue Register API parser
+$apiParser = new CodecheckRegisterGithubIssuesApiParser();
+
+// CODECHECK Register with list of all identifiers in range
+$certificateIdentifierList = CertificateIdentifierList::fromApi($apiParser);
+
+// print Certificate Identifier list
+$certificateIdentifierList->sortDesc();
+echo $certificateIdentifierList->toStr();
+
+echo $certificateIdentifierList->getNewestIdentifier()->toStr() . "\n";
+
+$new_identifier = CertificateIdentifier::newUniqueIdentifier($certificateIdentifierList);
+
+$codecheckVenueTypes = new CodecheckVenueTypes();
+$codecheckVenueNames = new CodecheckVenueNames();
```

Comment:
Please create an issue for when integrating this with the plugin that these requests need to be cached somehow. Check if there is a mechanism in OJS to store this information easily, e.g., as a JSON object in a hidden property field of the plugin.